src/query-compiler/default-query-compiler.ts

```diff
@@ -1428,7 +1428,13 @@ export class DefaultQueryCompiler
       this.append('if exists ')
     }
 
-    this.visitNode(node.name)
+    this.compileList(node.names)
+
+    if (node.modifier === 'cascade') {
+      this.append(' cascade')
+    } else if (node.modifier === 'restrict') {
+      this.append(' restrict')
+    }
```

Comment:
It's OK to trust internals that create the node here.


```suggestion
    if (node.modifier) {
      this.append(` ${node.mofider}`)
    }
```